javascript/widgets/handler/passwordsignup.js

```diff
@@ -97,14 +105,19 @@ firebaseui.auth.widget.handler.onSignUpSubmit_ = function(app, component) {
           ),
       [email, password],
       function(user) {
-        // Sign up successful. We can now set the name.
-        return app.registerPending(user.updateProfile({'displayName': name})
-            .then(function() {
-              // Pass password credential to complete the sign-in to original
-              // auth instance.
-              firebaseui.auth.widget.handler.common.setLoggedIn(
-                  app, component, emailPassCred);
-            }));
+        if (requireDisplayName) {
+          // Sign up successful. We can now set the name.
+          return app.registerPending(user.updateProfile({'displayName': name})
+              .then(function() {
+                // Pass password credential to complete the sign-in to original
+                // auth instance.
+                firebaseui.auth.widget.handler.common.setLoggedIn(
+                    app, component, emailPassCred);
+              }));
+        } else {
```

Comment:
There are no tests for this. Please add tests in passwordsignup_test.js when display name is not required.